src/vs/base/parts/sandbox/electron-browser/preload.js

```diff
@@ -238,28 +240,33 @@
 	 * @param {{[key: string]: string}} userEnv
 	 * @returns {Promise<void>}
 	 */
-	function resolveEnv(userEnv) {
-		if (!resolvedEnv) {
+	async function resolveEnv(userEnv) {
+		if (!shellEnv) {
 
 			// Apply `userEnv` directly
 			Object.assign(process.env, userEnv);
 
 			// Resolve `shellEnv` from the main side
-			resolvedEnv = new Promise(function (resolve) {
-				ipcRenderer.once('vscode:acceptShellEnv', function (event, shellEnv) {
-
-					// Assign all keys of the shell environment to our process environment
-					// But make sure that the user environment wins in the end
-					Object.assign(process.env, shellEnv, userEnv);
-
-					resolve();
+			shellEnv = new Promise(function (resolve) {
+				ipcRenderer.once('vscode:acceptShellEnv', function (event, shellEnvResult) {
+					if (process.env['VSCODE_IGNORE_SHELL_ENV'] /* TODO@bpasero for https://github.com/microsoft/vscode/issues/108804 */) {
```

Comment:
I suggest something more contextual to the behavior like `VSCODE_SKIP_PROCESS_ENV_PATCHING`, since it's not about ignoring the shell env, but skipping patching it to the `process.env`.